src/pipeline/unix_pipe.rs

```diff
@@ -43,8 +43,10 @@ pub trait UnixPipeImpl: Sized {
     fn new(fd: RawFd) -> Result<Self>;
```

Comment:
Nit: It would be more idiomatic to call the trait `UnixPipe` and the type alias `UnixFile`.